features/device-store/org.wso2.carbon.device.mgt.enterprise.store.feature/src/main/resources/publisher/samples/assets/deviceTypes/install.js

```diff
@@ -0,0 +1,164 @@
+		artifact.attributes.images_banner = context.assetPath + artifact.attributes.images_banner;
+		artifact.attributes.overview_url = artifact.attributes.overview_url;
+		artifact.attributes.overview_createdtime = now;
+		//artifact.attributes.images_thumbnail = context.httpContext + artifact.attributes.images_thumbnail;
+		//artifact.attributes.images_banner = context.httpContext + artifact.attributes.images_banner;
+
+		artifact.type = 'deviceType';
+
+		//Create the deployment object
+		context['artifact'] = artifact;
+
+
+		//Set the tags
+		context['tags'] = jsonConfig.tags.split(',');
+
+		//Set the ratings
+		context['rate'] = jsonConfig.rate;
+		context['path'] = '/_system/governance/' + context.assetType + '/' + artifact.attributes.overview_provider +
+			'/' + artifact.attributes.overview_name + '/' + artifact.attributes.overview_version;
+
+		context['type'] = 'deviceType';
+		log.debug('tags located: ' + context.tags);
+		log.debug('rate located: ' + context.rate);
+	}
+
+	/*
+	 The function attaches a lifecycle to the asset
+	 */
+	function onAttachLifecycle(context) {
+		var artifactManager = context.artifactManager;
+		var currentAsset = context.currentAsset;
+		var currentLifeCycleName = currentAsset.lifecycle || null;
+		var currentLifeCycleState = currentAsset.lifecycleState || null;
+		var attempts = 0;
+
+		log.debug('attaching lifecycle to: ' + currentAsset.attributes.overview_name);
+
+		log.debug('current lifecycle: ' + currentLifeCycleName + ' , current state: ' + currentLifeCycleState);
+
+
+		//Check if a lifecycle has been attached
+		if (!currentLifeCycleName) {
+
+			log.debug('before calling current asset ' + DEFAULT_LIFECYCLE);
+			log.debug(currentAsset);
+
+			//Attach the lifecycle
+			artifactManager.attachLifecycle(DEFAULT_LIFECYCLE, currentAsset);
+
+			//Update the current asset
+			currentAsset = artifactManager.get(currentAsset.id);
+		}
+		else {
+			//We skip moving to the Published state.
+			log.debug('skipping promotion operations as a lifecycle has been attached');
+			return;
+		}
+
+		log.debug('attempting to execute the robot path: '+stringify(pathToDesiredState));
+
+		executeRobotPath(pathToDesiredState,artifactManager,currentAsset);
+
+		log.debug('finished executing robot path');
+
+		//Try to reach the desired life-cycle state before the attempt limit
+		/*while ((currentLifeCycleState != DESIRED_LIFECYCLE_STATE) && (attempts < PROMOTE_COUNT)) {
+
+		 artifactManager.promoteLifecycleState(INVOKED_OPERATION, currentAsset);
+
+		 //Update the current life-cycle state.
+		 currentLifeCycleState = artifactManager.getLifecycleState(currentAsset);
+
+		 //Increase the attempts by one
+		 attempts++;
+
+		 log.debug('current lifecycle state: ' + currentLifeCycleState);
+		 }
+
+		 log.debug('final state of : ' + currentAsset.attributes.overview_name + ' ' + currentLifeCycleState);  */
+	}
+
+	function executeRobotPath(path,artifactManager,asset){
```

Comment:
Acknowledge, this code was belongs to IoT server ES extension app which is deprecated after moving to UUF based devicemgt app. So unable to fix the formatting issue.